lib/arke/strategy/base.rb

```diff
@@ -0,0 +1,71 @@
+require 'exchange'
+
+module Arke::Strategy
+  class Base
+    RunNotOverriden = Class.new(StandardError)
+
+    def initialize
+      @shutdown = false
+      # we limit strategy update rate here (seconds)
+      # easy to change if we will implement some stress-testing strategies
+      @minimum_update_period = Arke::Configuration.get(:min_update_period) || 1.0
+    end
+
+    def start
+      sources = Arke::Configuration.require!(:sources).collect do |source|
+        Arke::Exchange.create(source['driver'], self)
+      end
+
+      target = Arke::Exchange.create(Arke::Configuration.require!(:target)['driver'], self)
+
+      @threads = sources.collect { |source| Thread.new {source.start } }
+      @threads.push(Thread.new { target.start })
+
+      trap('SIGINT') { stop }
+
```

Comment:
You will need to change this later, also it would be nice to hide thread creation from developers, we will think of something.

Your code seems weird you abuse of collect and lack of clarity, you code for people to read it and understand at a glance.

Better use a hash and a car for threads
def target
  @workers[:target]
end